src/main/scala/io/iohk/ethereum/blockchain/sync/FastSync.scala

```diff
@@ -222,11 +224,20 @@ trait FastSync {
         cleanupRequestedMaps(ref)
 
       case PrintStatus =>
-        val totalNodesCount = downloadedNodesCount + mptNodesQueue.size + nonMptNodesQueue.size
-        log.info(
-          s"""|Block: ${appStateStorage.getBestBlockNumber()}/${initialSyncState.targetBlock.number}.
-              |Peers: ${assignedHandlers.size}/${handshakedPeers.size} (${blacklistedPeers.size} blacklisted).
-              |State: $downloadedNodesCount/$totalNodesCount known nodes.""".stripMargin.replace("\n", " "))
+        printStatus()
+    }
+
+    private def printStatus() = {
+      val totalNodesCount = downloadedNodesCount + mptNodesQueue.size + nonMptNodesQueue.size
+      val formatPeer: (Peer) => String = peer => s"${peer.remoteAddress.getAddress.getHostAddress}:${peer.remoteAddress.getPort}"
+      log.info(
+        s"""|Block: ${appStateStorage.getBestBlockNumber()}/${initialSyncState.targetBlock.number}.
+            |Peers waiting_for_response/connected: ${assignedHandlers.size}/${handshakedPeers.size} (${blacklistedPeers.size} blacklisted).
+            |State: $downloadedNodesCount/$totalNodesCount nodes.
+            |Details: connected(${assignedHandlers.values.map(formatPeer).toSeq.sorted})/
+            |handshaked(${handshakedPeers.keys.map(formatPeer).toSeq.sorted})
+            | blacklisted(${blacklistedPeers.map { case (id, _) => id }})
```

Comment:
Very minor detail: Maybe it would be better to print out `id.value` and use `mkString(", ")` and do something like `blacklistedPeers.map { case (id, _) => id.value }.mkString(", ")`, so as to not print the use of `List` and `PeerId`?
We could also do something similar with `connected` and `handshaked`.